src/conn.rs

```diff
@@ -374,6 +377,7 @@ impl<'a, 'c> InProgress<'a, 'c> {
                            self.cache.transact_watcher(),
                            terms,
                            tempid_set)?;
+        self.tx_reports.push(report.clone());
```

Comment:
Can we make `transact_terms` return `Result<&TxReport>` now, avoiding the clone here? Or do the lifetimes not line up?

Reply:
I can't get the lifetimes to work here.

Reply:
Another idea to consider, then: don't return the report, return the `txid`. The caller can retrieve it from `in_progress.tx_reports` if they want it. It's an API change, but I think often the caller just wants success/failure, not the whole report.